app/org/sagebionetworks/bridge/services/ParticipantService.java

```diff
@@ -280,40 +320,29 @@ public StudyParticipant getParticipant(Study study, Account account, boolean inc
         builder.withHealthCode(account.getHealthCode());
         builder.withClientData(account.getClientData());
         builder.withAttributes(account.getAttributes());
-        SubstudyAssociations assoc = BridgeUtils.substudyAssociationsVisibleToCaller(account.getAccountSubstudies());
         builder.withSubstudyIds(assoc.getSubstudyIdsVisibleToCaller());
         builder.withExternalIds(assoc.getExternalIdsVisibleToCaller());
-
-        if (includeHistory) {
-            Map<String,List<UserConsentHistory>> consentHistories = Maps.newHashMap();
-            // The history includes all subpopulations whether they match the user or not.
-            List<Subpopulation> subpopulations = subpopService.getSubpopulations(study.getStudyIdentifier(), false);
-            for (Subpopulation subpop : subpopulations) {
-                // always returns a list, even if empty
-                List<UserConsentHistory> history = getUserConsentHistory(account, subpop.getGuid());
-                consentHistories.put(subpop.getGuidString(), history);
-            }
-            builder.withConsentHistories(consentHistories);
-            // To calculate consent status, we need construct a CriteriaContext from RequestInfo (only for the 
-            // information we don't store in account).
-            RequestInfo requestInfo = cacheProvider.getRequestInfo(account.getId());
-            if (requestInfo != null) {
-                CriteriaContext criteriaContext = new CriteriaContext.Builder()
-                        .withStudyIdentifier(study.getStudyIdentifier())
-                        .withUserId(account.getId())
-                        .withHealthCode(account.getHealthCode())
-                        .withClientInfo(requestInfo.getClientInfo())
-                        .withLanguages(requestInfo.getLanguages())
-                        .withUserDataGroups(account.getDataGroups())
-                        .withUserSubstudyIds(assoc.getSubstudyIdsVisibleToCaller())
-                        .build();
-                Map<SubpopulationGuid, ConsentStatus> consentStatusMap = consentService.getConsentStatuses(
-                        criteriaContext, account);
-                boolean isConsented = ConsentStatus.isUserConsented(consentStatusMap);
-                builder.withConsented(isConsented);
-            }
-        }
-        return builder.build();
+        return builder;
+    }
+    
+    private StudyParticipant.Builder addHistory(StudyParticipant.Builder builder, Account account, CriteriaContext context) {
+        Map<String,List<UserConsentHistory>> consentHistories = Maps.newHashMap();
+        // The history includes all subpopulations whether they match the user or not.
+        List<Subpopulation> subpopulations = subpopService.getSubpopulations(context.getStudyIdentifier(), false);
```

Comment:
Context is not needed, except for studyId. Would it be better to just take studyId instead?

Reply:
Removed this. Once you do this, you can reorganize the code to avoid creating CriteriaContext unless you are asking for history.